packages/nuxt/src/plugins/plugin.ts

```diff
@@ -226,16 +227,10 @@ function getOutputs(
 ): {
   buildOutputs: string[];
 } {
-  let nuxtBuildDir = nuxtConfig?.buildDir;
-  if (nuxtConfig?.buildDir && basename(nuxtConfig?.buildDir) === '.nuxt') {
-    // if buildDir exists, it will be `something/something/.nuxt`
-    // we want the "general" outputPath to be `something/something`
-    nuxtBuildDir = nuxtConfig.buildDir.replace(
-      basename(nuxtConfig.buildDir),
-      ''
-    );
-  }
-  const buildOutputPath = normalizeOutputPath(nuxtBuildDir, projectRoot);
+  const buildOutputPath = normalizeOutputPath(
```

Comment:
We actually only want to cache the `.nuxt` folder. This is consistent with Next.js where we cache only `.next` folder. Previously, this logic was to handle caching `dist/<app>` rather than just `dist/<app>/.nuxt`, but we've since moved the default `buildDir` back to within the project root anyway. We cannot cache the entire `{projectRoot}`, which includes source code.